src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.cpp

```diff
@@ -0,0 +1,77 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace trajectory
+{
+
+CoordinatesBroker::CoordinatesBroker()
+    : nextCoordinatesSubsetIndex_(0),
+      coordinatesSubsetIndexMap_({})
+{
+}
+
+CoordinatesBroker::CoordinatesBroker(const Array<CoordinatesSubset>& aCoordinatesSubsetArray)
+    : nextCoordinatesSubsetIndex_(0),
+      coordinatesSubsetIndexMap_({})
+{
+    for (CoordinatesSubset coordinatesSubset : aCoordinatesSubsetArray)
+    {
+        this->addSubset(coordinatesSubset);
+    }
+}
+
+Index CoordinatesBroker::addSubset(const CoordinatesSubset& aCoordinatesSubset)
+{
+    auto search = this->coordinatesSubsetIndexMap_.find(aCoordinatesSubset);
```

Comment:
```suggestion
    const auto search = this->coordinatesSubsetIndexMap_.find(aCoordinatesSubset);
```

I would make the name more explicit for `search`, like `coordinatesSubsetSearchIndex`